docs/remote/create-dev-container.md

```diff
@@ -56,253 +43,162 @@ All of the predefined container configurations you can pick from come from the [
 
-```json
-"mounts": [
-    "source=${localEnv:HOME}${localEnv:USERPROFILE},target=/host-home-folder,type=bind,consistency=cached"
-]
+```bash
+# If sudo is installed and configured
+sudo apt-get update
+# Install Git
+sudo apt-get install git
 ```
 
-You can also reference `"${localWorkspaceFolder}` if you need to mount something from the local filesystem into the container.
-
-The `runArgs` property supports the same list of arguments as the [docker run](https://docs.docker.com/engine/reference/commandline/run/) command and can be useful for a wide variety of scenarios including [setting environment variables](/docs/remote/containers-advanced.md#adding-environment-variables).
+### Rebuilding
 
-If your application was built using C++, Go, or Rust, or another language that uses a **ptrace-based debugger**, the `runArgs` property can also be used to configure needed runtime security and capability settings.
+When editing the contents of the `.devcontainer` folder, you'll need to rebuild for changes to take effect. Use the **Remote-Containers: Rebuild Container** command for your container to update.
 
-For example:
+However, if you **rebuild** the container, you will have to **reinstall** anything you've installed manually. To avoid this problem, you can use a series of commands in the `postCreateCommand` property in `devcontainer.json`:
 
 ```json
 {
-    "name": "My Go App",
-    "dockerFile": "Dockerfile",
+    "image": "mcr.microsoft.com/vscode/devcontainers/typescript-node:0-12",
     "extensions": [
-        "golang.go"
+        "dbaeumer.vscode-eslint"
     ],
-    "runArgs": [
-        "--cap-add=SYS_PTRACE",
-        "--security-opt",
-        "seccomp=unconfined"
-    ]
+    "forwardPorts": [ 3000 ],
+    "postCreateCommand": "sudo apt-get update && sudo apt-get install git",
 }
 ```
 
-While less efficient than a custom Dockerfile, you can also use the `postCreateCommand` property to **install additional software** that **may not be in your base image** or for cases where you would prefer **not to modify a deployment Dockerfile** you are reusing.
+The `postCreateCommand` is run once the container is running, so you can also use the property to run commands like `npm install` or to execute a shell script in your source tree (if you have mounted it).
+
+```json
+"postCreateCommand": "bash scripts/install-dev-tools.sh"
+```
+
+Rather than referencing an image directly in `devcontainer.json` or installing software via the `postCreateCommand`, an even more efficient practice is to use a Dockerfile.
 
-For example, here is a `devcontainer.json` that adds compiler tools and the [C++](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cpptools) extension to the base Ubuntu 18.04 container image:
+## Dockerfile
+
+A Dockerfile will also live in the `.devcontainer` folder. You can replace the `image` property in `devcontainer.json` with `dockerFile`:
 
 ```json
 {
-    "image": "ubuntu:18.04",
+    "dockerFile": "Dockerfile",
```

Comment:
I'm thinking of all sorts of things re-reading this. 😀  I wonder if we should use the new "build": { "dockerfile": "Dockerfile" } syntax here.  That's where things like "args" live.  A bit more verbose, but would set people up for success later.

Reply:
I agree updating to the new syntax would make sense to set readers up for success later! Updated this reference- I didn't see other outdated references, but if I missed any, please just let me know.